src/core/Akka.Persistence/Journal/AsyncWriteJournal.cs

```diff
@@ -247,23 +247,25 @@ private void HandleReplayMessages(ReplayMessages message)
 
             var readHighestSequenceNrFrom = Math.Max(0L, message.FromSequenceNr - 1);
             var promise = new TaskCompletionSource<long>();
-            _breaker.WithCircuitBreaker(() => ReadHighestSequenceNrAsync(message.PersistenceId, readHighestSequenceNrFrom))
-                .ContinueWith(t =>
+
+            async Task ExecuteHighestSequenceNr()
+            {
+                try
                 {
-                    if (!t.IsFaulted && !t.IsCanceled)
+                    var highSequenceNr = await _breaker.WithCircuitBreaker(() =>
+                        ReadHighestSequenceNrAsync(message.PersistenceId, readHighestSequenceNrFrom));
```

Comment:
Should we consider adding a `WithCircuitBreaker<TState,TResult>` overload to the circuit breakers to get rid of an allocation here? (Pass `this` and `message` in, get vaue of `readHighestSequenceNrFrom` inside the function:)

```
var highSequenceNr = await _breaker.WithCircuitBreaker((this,message),(state) =>
                        ReadHighestSequenceNrAsync(message.PersistenceId, Math.Max(0L, message.FromSequenceNr - 1)));
```

That should? actually also get rid of one local on top of the closure allocation in question.

OTOH, this is the read side, which isn't as hot as the write side.

Reply:
I think that'd be a good idea - with this PR, so far, I wanted to make the changes gradually so I've been very minimalist about what I touch downstream.